Configs/.config/hypr/scripts/cpuinfo.sh

```diff
@@ -1,35 +1,80 @@
-#!/usr/bin/env sh
+#!/bin/bash
 
-# CPU model
-model=$(lscpu | awk -F ':' '/Model name/ {sub(/^ *| *$/,"",$2); print $2}' | awk '{NF-=3}1')
 
-# CPU utilization
-utilization=$(top -bn1 | awk '/^%Cpu/ {print 100 - $8}')
+# Thee shalt find the greatest one,
+# He who not more than the chosen one
+map_floor() {
 
-# Clock speed
-freqlist=$(cat /proc/cpuinfo | grep "cpu MHz" | awk '{ print $4 }')
-maxfreq=$(lscpu | grep "CPU max MHz" | awk -F: '{ print $2}' | sed -e 's/ //g' -e 's/\.[0-9]*//g')
-frequency=$(echo $freqlist | tr ' ' '\n' | awk "{ sum+=\$1 } END {printf \"%.0f/$maxfreq MHz\", sum/NR}")
+    # From the depths of the string, words arise,
+    # Keys in pairs, a treasure in disguise.
+    IFS=', ' read -r -a pairs <<< "$1"
 
-# CPU temp
-temp=$(sensors | awk '/Package id 0/ {print $4}' | awk -F '[+.]' '{print $2}')
-if [ -z "$temp" ]; then
-    temp=$(sensors | awk '/Tctl/ {print $2}' | tr -d '+°C')
-fi
-if [ -z "$temp" ]; then
-    temp="N/A"
-fi
+    # If the final token stands alone and bold,
+    # Declare it the default, its worth untold.
+    if [[ ${pairs[-1]} != *":"* ]]; then
+        def_val="${pairs[-1]}"
+        unset 'pairs[${#pairs[@]}-1]'
+    fi
+
+    # Scans the map, a peak it seeks,
+    # The highest passed, the value speaks.
+    for pair in "${pairs[@]}"; do
+        IFS=':' read -r key value <<< "$pair"
+        
+        # Behold! Thou holds the secrets they seek,
+        # Declare it and silence the whispers unique.
+        if [ ${2%%.*} -gt $key ]; then
+            echo "$value"
+            return
+        fi
+    done
 
-# map icons
-set_ico="{\"thermo\":{\"0\":\"\",\"45\":\"\",\"65\":\"\",\"85\":\"\"},\"emoji\":{\"0\":\"❄\",\"45\":\"☁\",\"65\":\"\",\"85\":\"\"},\"util\":{\"0\":\"󰾆\",\"30\":\"󰾅\",\"60\":\"󰓅\",\"90\":\"\"}}"
-eval_ico() {
-    map_ico=$(echo "${set_ico}" | jq -r --arg aky "$1" --argjson avl "$2" '.[$aky] | keys_unsorted | map(tonumber) | map(select(. <= $avl)) | max')
-    echo "${set_ico}" | jq -r --arg aky "$1" --arg avl "$map_ico" '.[$aky] | .[$avl]'
+    # On this lonely shore, where silence dwells
+    # Even the waves, echoes words unheard
+    [ -n "$def_val" ] && echo $def_val || echo " "
 }
 
-thermo=$(eval_ico thermo $temp)
-emoji=$(eval_ico emoji $temp)
-speedo=$(eval_ico util $utilization)
+# Define glyphs
+if [[ $NO_EMOJI -eq 1 ]]; then
+    temp_lv="85:, 65:, 45:☁, ❄"
+else
+    temp_lv="85:🌋, 65:🔥, 45:☁️, ❄️"
+fi
+util_lv="90:, 60:󰓅, 30:󰾅, 󰾆" 
+
+# Get static CPU information
+model=$(lscpu | awk -F': ' '/Model name/ {gsub(/^ *| *$| CPU.*/,"",$2); print $2}')
+maxfreq=$(lscpu | awk '/CPU max MHz/ { sub(/\..*/,"",$4); print $4}')
+
+# Get CPU stat
+statFile=$(cat /proc/stat | head -1)
+prevStat=$(awk '{print $2+$3+$4+$6+$7+$8 }' <<< $statFile)
+prevIdle=$(awk '{print $5 }' <<< $statFile)
+
+while true; do
```

Comment:
why the loop? its not recommended to run this as a daemon process.

Reply:
> ## Continuous script
> 
> As the script has to recalculate values that doesn't change during runtime (like CPU name and max freq), it is inefficient to run on every 5s. Since waybar allows the scripts to be executed in continuous mode which allows significant improvement on performance as the static values doesn't required to be recalculated on each iteration, the cpuinfo.sh has been adjusted to run as a continuous script.

It can be made to execute once in every 5s instead by saving the prev statistics to a temp file. However, I don't see why it is not recommended to run it as a daemon as re-executing the script has to deal with unnecessary system calls, read-writes and re-interpreting the script once in every 5s.

Also, it is not going to save cpu time on the scheduler as it has to deal with the sleep syscall from the script otherwise the same from the waybar.

It is also noted that the waybar builtin modules which requires updates on periodic intervals ([memory](https://github.com/Alexays/Waybar/blob/601af3de81acc0dbe5aa967d7f10f8a69d10bd02/src/modules/memory/common.cpp#L5-L8), [cpu_usage](https://github.com/Alexays/Waybar/blob/601af3de81acc0dbe5aa967d7f10f8a69d10bd02/src/modules/cpu_usage/common.cpp#L14C1-L17C5)) are running on separate daemon threads.

Reply:
> > ## Continuous script
> > As the script has to recalculate values that doesn't change during runtime (like CPU name and max freq), it is inefficient to run on every 5s. Since waybar allows the scripts to be executed in continuous mode which allows significant improvement on performance as the static values doesn't required to be recalculated on each iteration, the cpuinfo.sh has been adjusted to run as a continuous script.
> 
> It can be made to execute once in every 5s instead by saving the prev statistics to a temp file. However, I don't see why it is not recommended to run it as a daemon as re-executing the script has to deal with unnecessary system calls, read-writes and re-interpreting the script once in every 5s.

Oh my bad I missed it, you did mention it, but it was too long so I didn't read :(
But still my concern is valid... I do not use the any of these monitoring modules in waybar as i often switch the waybar mode... So my question is that, will this process still run when the waybar mode changes?

Reply:
As per the `confgen.sh` architecture, the waybar configuration file `config.jsonc` is regenerated on switching the waybar mode. So this process won't be summoned by new instances of waybar as the modules will not be present in the `config.jsonc`.

https://github.com/prasanthrangan/hyprdots/blob/8369bedd86a8f5ea9687b8d23dc7ee64faf83ff2/Configs/.local/share/bin/wbarconfgen.sh#L125-L130
According the above snippet, when you switch the mode, the waybar and its entire process tree will be killed, including our poor little daemon.

Sorry for the late reply, was little too busy these days.